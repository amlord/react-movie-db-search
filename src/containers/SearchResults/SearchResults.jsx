import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Wrapper from 'components/Wrapper';
import SearchBar from 'components/SearchBar';
import SearchResultsOverview from 'components/SearchResultsOverview';
import SearchResultsList from 'components/SearchResultsList';
import Pagination from 'components/Pagination';

const StyledMain = styled.main`
  background-color: #fff;
  color: #333;
`;

const SearchResults = ({ results, totalResults, onSearchUpdate }) => (
  <StyledMain>
    <Wrapper>
      <SearchBar onSearchUpdate={onSearchUpdate} />
      <SearchResultsOverview totalResults={totalResults} />
      <Pagination />
      <SearchResultsList results={results} />
      <Pagination />
    </Wrapper>
  </StyledMain>
);

SearchResults.defaultProps = {
  results: [],
  totalResults: 0,
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object),
  totalResults: PropTypes.number,
  onSearchUpdate: PropTypes.func.isRequired,
};

export default SearchResults;
