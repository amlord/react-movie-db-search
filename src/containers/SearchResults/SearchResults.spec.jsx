import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults';

describe('SearchResults', () => {
  it('displays correctly', () => {
    const props = {
      results: [],
      onSearchUpdate: jest.fn(),
    };
    const wrapper = shallow(<SearchResults {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
