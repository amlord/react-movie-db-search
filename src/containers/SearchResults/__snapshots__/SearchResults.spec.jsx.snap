// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    onSearchUpdate={[MockFunction]}
    results={Array []}
    totalResults={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Wrapper>
        <SearchBar
          onSearchUpdate={[MockFunction]}
        />
        <SearchResultsOverview
          totalResults={0}
        />
        <Pagination />
        <SearchResultsList
          results={Array []}
        />
        <Pagination />
      </Wrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SearchBar
            onSearchUpdate={[MockFunction]}
          />,
          <SearchResultsOverview
            totalResults={0}
          />,
          <Pagination />,
          <SearchResultsList
            results={Array []}
          />,
          <Pagination />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSearchUpdate": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "totalResults": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "results": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Wrapper>
          <SearchBar
            onSearchUpdate={[MockFunction]}
          />
          <SearchResultsOverview
            totalResults={0}
          />
          <Pagination />
          <SearchResultsList
            results={Array []}
          />
          <Pagination />
        </Wrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <SearchBar
              onSearchUpdate={[MockFunction]}
            />,
            <SearchResultsOverview
              totalResults={0}
            />,
            <Pagination />,
            <SearchResultsList
              results={Array []}
            />,
            <Pagination />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onSearchUpdate": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "totalResults": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "results": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
